kevm-pyk/src/tests/integration/test_foundry_prove.py

```diff
@@ -206,6 +206,32 @@ def test_foundry_auto_abstraction(foundry_root: Path, update_expected_output: bo
     assert_or_update_show_output(show_res, TEST_DATA_DIR / 'gas-abstraction.expected', update=update_expected_output)
 
 
+def test_foundry_remove_node(foundry_root: Path, update_expected_output: bool) -> None:
+    prove_res = foundry_prove(
+        foundry_root,
+        tests=['AssumeTest.test_assume_true'],
+    )
+    assert_pass('AssumeTest.test_assume_true', prove_res)
+
+    foundry_remove_node(
+        foundry_root=foundry_root,
+        test='AssumeTest.test_assume_true',
+        node=4,
+    )
+    prove_res = foundry_prove(
+        foundry_root,
+        tests=['AssumeTest.test_assume_true'],
+        max_depth=0,
+    )
+    assert_fail('AssumeTest.test_assume_true', prove_res)
```

Comment:
Why do we need the extra call to `foundry_prove` here? Can't we just assert that the proof is now pending?

Reply:
The foundry_x level functions don't actually return the proof. This was the only way I see to get the proof result. I will refactor this in a future PR.